src/protocol/context/mod.rs

```diff
@@ -0,0 +1,94 @@
+use crate::ff::Field;
+use crate::helpers::messaging::{Gateway, Mesh};
+use crate::helpers::Role;
+use crate::protocol::mul::SecureMul;
+use crate::protocol::prss::{
+    Endpoint as PrssEndpoint, IndexedSharedRandomness, SequentialSharedRandomness,
+};
+use crate::protocol::reveal::Reveal;
+use crate::protocol::{Step, Substep};
+use crate::secret_sharing::SecretSharing;
+use std::sync::Arc;
+
+mod malicious;
+mod semi_honest;
+
+pub use malicious::MaliciousProtocolContext;
+pub use semi_honest::SemiHonestProtocolContext;
+
+/// Context used by each helper to perform secure computation. Provides access to shared randomness
+/// generator and communication channel.
+pub trait ProtocolContext<F: Field>:
+    Clone + SecureMul<F, Share = <Self as ProtocolContext<F>>::Share> + Reveal
```

Comment:
```suggestion
    Clone + SecureMul<F, Share = <Self as ProtocolContext<F>>::Share> + Reveal<Share = <Self as ProtocolContext<F>>::Share>
```

Reveal and SecureMul are different in that one takes F and the other does not.  Can we work out which one we prefer?  If we can avoid multiple type parameters, maybe that is better.

Reply:
There was a need to make `Reveal` work with any field as `Context<F>` was used to reveal `Fp2` values. It is no longer the case, so I am going to change `Reveal` and make it look the same as `SecureMul`